### PR TITLE
Decrypt aspera@0.1.0

### PR DESCRIPTION
**Pull request by edsu7**

### CODE REVIEW

File: decrypt-aspera/main.nf [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+// tool specific parmas go here, add / change as needed
+params.input_file = ""
+params.output_pattern = "*"  // output file name pattern
+
+
+process decryptAspera {
+  container "${params.container ?: container[params.container_registry ?: default_container_registry]}:${params.container_version ?: version}"
+  publishDir "${params.publish_dir}/${task.process.replaceAll(':', '_')}", mode: "copy", enabled: params.publish_dir
```
Comment (lindaxiang) [inline decrypt-aspera/main.nf]: Please add ` ? true : false` at the end. 
```
publishDir "${params.publish_dir}/${task.process.replaceAll(':', '_')}", mode: "copy", enabled: params.publish_dir ? true : false
```

File: decrypt-aspera/main.nf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+params.container = ""
+
+params.cpus = 1
+params.mem = 1  // GB
+params.publish_dir = ""  // set to empty string will disable publishDir
+
+
+// tool specific parmas go here, add / change as needed
+params.input_file = ""
+params.output_pattern = "*"  // output file name pattern
```
Comment (lindaxiang) [inline decrypt-aspera/main.nf]: can remove it if this is not needed.

File: decrypt-aspera/main.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def main():
+    """
+    Python implementation of tool: decrypt-aspera
+
+    This is auto-generated Python code, please update as needed!
+    """
+    parser = argparse.ArgumentParser(description='differentiate JSON metadata payload for SONG upload')
+    parser.add_argument('-f', '--file', dest="file", help="auto generated json", required=True)
```
Comment (lindaxiang) [inline decrypt-aspera/main.py]: Need to update description?

File: decrypt-aspera/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Package decrypt-aspera
+
+
+Please update this with a brief introduction of the package.
```
Comment (lindaxiang) [inline decrypt-aspera/README.md]: The `README.md` file needs to update.

File: decrypt-aspera/main.nf [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    """
+}
+
+// this provides an entry point for this main script, so it can be run directly without clone the repo
+// using this command: nextflow run <git_acc>/<repo>/<pkg_name>/<main_script>.nf -r <pkg_name>.v<pkg_version> --params-file xxx
+workflow {
+  decryptAspera(
+    params.file,
+    params.c4gh_secret_key
```
Comment (lindaxiang) [inline decrypt-aspera/main.nf]: I need to change to the following to get it work:
```
file(params.file),
file(params.c4gh_secret_key)
```

File: decrypt-aspera/main.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    results = parser.parse_args()
+    
+
+    cmd="cat "+results.file+" | crypt4gh decrypt > "+re.sub('.c4gh$','',results.file)
```
Comment (lindaxiang) [inline decrypt-aspera/main.py:43]: module `re` is not imported

File: decrypt-aspera/main.nf (hunk 1 above)
Comment (lindaxiang) [inline decrypt-aspera/main.nf]: Please add ` ? true : false` at the end
```
publishDir "${params.publish_dir}/${task.process.replaceAll(':', '_')}", mode: "copy", enabled: params.publish_dir ? true : false
```

**Review by lindaxiang — CHANGES_REQUESTED** [review]

File: decrypt-aspera/tests/checker.nf [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+params.cpus = 1
+params.mem = 1  // GB
+params.publish_dir = ""  // set to empty string will disable publishDir
+
+// tool specific parmas go here, add / change as needed
+params.input_file = ""
+params.expected_output = "./expected/mystery_contents.txt"
+
+process decryptAspera {
```
Comment (lindaxiang) [inline decrypt-aspera/tests/checker.nf]: why not import the process
```
include { decryptAspera } from '../main'
```

File: decrypt-aspera/tests/checker.nf [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  take:
+    input_file
+    c4gh_secret_key
+    expected_output
+
+  main:
+    decryptAspera(
+    input_file,
+    params.c4gh_secret_key
+   )
```
Comment (lindaxiang) [inline decrypt-aspera/tests/checker.nf:94]: I need to change to the following to get it work:
```
file(input_file),
file(c4gh_secret_key)
```

File: decrypt-aspera/tests/checker.nf [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+/********************************************************************/
+/* this block is auto-generated based on info from pkg.json where   */
+/* changes can be made if needed, do NOT modify this block manually */
+nextflow.enable.dsl = 2
+version = '0.1.0'  // package version
+
+container = [
+    'ghrc.io': 'ghrc.io/icgc-argo/argo-data-submission.decrypt-aspera'
+]
+default_container_registry = 'ghrc.io'
```
Comment (lindaxiang) [inline decrypt-aspera/tests/checker.nf]: Spent quite a while found this typo: `ghrc.io` should be `ghcr.io`.
This part of code should be auto-generated, not sure how it happened. 
Did you manually change it? :-)

**Review by lindaxiang — CHANGES_REQUESTED** [review]

File: decrypt-aspera/main.nf (hunk 2 above)
Comment (lindaxiang) [reply decrypt-aspera/main.nf]: change `params.input_file` to `params.file`?
it's better to also initialize `params.c4gh_secret_key` here

File: decrypt-aspera/main.nf (hunk 1 above)
Comment (edsu7) [reply decrypt-aspera/main.nf]: Added your suggested fix.
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/main.nf#L53

File: decrypt-aspera/main.nf (hunk 2 above)
Comment (edsu7) [reply decrypt-aspera/main.nf]: Initialized the two suggest params
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/main.nf#L47-L48

File: decrypt-aspera/main.py (hunk 3 above)
Comment (edsu7) [reply decrypt-aspera/main.py]: Added up to description
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/main.py#L37-L38

File: decrypt-aspera/README.md (hunk 4 above)
Comment (edsu7) [reply decrypt-aspera/README.md]: Updated [README.md](https://github.com/icgc-argo/argo-data-submission/blob/decrypt-aspera%400.1.0/decrypt-aspera/README.md) with brief description and instructions on how to run

File: decrypt-aspera/main.nf (hunk 5 above)
Comment (edsu7) [reply decrypt-aspera/main.nf]: Gah. Thank you. I think I mentioned I could only get the tests working with absolute paths and I think this fixed it
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/main.nf#L79-L83

File: decrypt-aspera/main.py (hunk 6 above)
Comment (edsu7) [reply decrypt-aspera/main.py:43]: Sorry don't know how that got missed and even worked on my local build
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/main.py#L26

File: decrypt-aspera/main.nf (hunk 1 above)
Comment (edsu7) [reply decrypt-aspera/main.nf]: Duplicate?
See [this thread](https://github.com/icgc-argo/argo-data-submission/pull/7#discussion_r899437275)

File: decrypt-aspera/tests/checker.nf (hunk 7 above)
Comment (edsu7) [reply decrypt-aspera/tests/checker.nf]: Added import
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/tests/checker.nf#L58

File: decrypt-aspera/tests/checker.nf (hunk 8 above)
Comment (edsu7) [reply decrypt-aspera/tests/checker.nf:94]: Added fixed
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/tests/checker.nf#L89-L92

File: decrypt-aspera/tests/checker.nf (hunk 9 above)
Comment (edsu7) [reply decrypt-aspera/tests/checker.nf]: No idea what happened. I corrected the typos
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/tests/checker.nf#L39-L42

**Review by lindaxiang — APPROVED** [review]

**Pull request merged**